processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
Does this impose any performance overhead in the case where `postFilters` is empty? If so, we could get around that by having two different Cursor implementations – one with postFiltering and one without.